python/ray/air/util/tensor_extensions/utils.py

```diff
@@ -28,16 +30,52 @@ def _is_ndarray_variable_shaped_tensor(arr: np.ndarray) -> bool:
     return True
 
 
-def _create_possibly_ragged_ndarray(
+@PublicAPI(stability="alpha")
+def create_possibly_ragged_ndarray(
```

Comment:
An alternative would be exposing `create_strict_ragged_ndarray` instead (e.g. as `create_ragged_ndarray`), where the user only uses it if they know they're dealing with variable-shaped ndarrays. It would be an easier to understand API as well, with clear semantics ("create a ragged ndarray" rather than "create a possibly ragged ndarray") and a more narrow type spec (input could just be `Sequence[np.ndarray]`).